code/modules/modular_computers/file_system/programs/generic/uav.dm

```diff
@@ -0,0 +1,266 @@
+/obj/item/modular_computer
+	var/list/paired_uavs //Weakrefs, don't worry about it!
```

Comment:
This should definitely not be a computer-level variable. It should be on nano module level or, at most, at program level.

An easy alternative way to allow a nano module to connect to UAVs (since nano module is not a physical object you can click) would be to allow pairing with UAVs on directly adjacent tiles through the UI instead of requiring a click with computer inhand. That way, you open the program, move the UAV next to whatever computer you are running it on, and press a button in the UI to connect them. It also has a nice bonus of letting you control UAVs from devices which can not be picked up.

Reply:
I mean, that's great and all, but they would get lost every time you end that program.

Reply:
Fair point, in that case, i'd recommend going with program-level variable.

Reply:
Don't those also get destroyed when you end the program?

Reply:
Nope, nano modules are destroyed when program is ended, programs themselves are not. They are permanent (specifically, they are bound to the hard drive component).